docs/docs/internal-development/create-admin-components-with-theme-support.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Create admin-components with theme support
+---
+
+MUI components allow their styles & default props to be overridden using the theme.
+They also allow overriding of the styles of individual components using the `sx` prop.
+
+All Comet Admin components should support the same features.
+
+The basics of how to implement this can be found in the MUI docs: https://mui.com/material-ui/guides/creating-themed-components/
+
+## Examples of how you should be able to override components
+
+### Setting a component's default props
+
+E.g.: every instance of `MyComponent` should have a shadow by default.
+
+```ts
+const theme = createCometTheme({
+    components: {
+        CometAdminMyComponent: {
+            defaultProps: {
+                shadow: true,
+            },
+        },
+    },
+});
+```
+
+### Overriding the styles of a component
+
+E.g.: in every instance of `MyComponent`, the root should have a red border, and the header should be lime.
+
+```ts
+const theme = createCometTheme({
+    components: {
+        CometAdminMyComponent: {
+            styleOverrides: {
+                root: {
+                    borderColor: "red",
+                },
+                header: {
+                    backgroundColor: "lime",
+                },
+            },
+        },
+    },
+});
+```
+
+### Overriding the styles of a single instance of a component
+
+E.g.: in this instance of `MyComponent`, the root should have a red border, and the header should be lime.
+
+```tsx
+<MyComponent
+    sx={{ borderColor: "red" }}
+    slotProps={{ header: { sx: { backgroundColor: "lime" } } }}
+/>
+```
+
+## Example component
+
+```tsx
+import { CometColor } from "@comet/admin-icons";
+import { ComponentsOverrides, Typography } from "@mui/material";
+import { css, styled, SxProps, Theme, useThemeProps } from "@mui/material/styles";
+import React from "react";
+
+import { SlotProps } from "./helpers/SlotProps";
+
+/**
+ * The `{ComponentName}ClassKey` type defines all the individual elements (or slots) of a component or a modifier or alternative style of a component.
+ * - `root` is generally the outermost element of the component.
+ * - `header` in this case would be the header element of the component.
+ * - `hasShadow` would be a modifier class added to the root element when the `shadow` prop is set to `true`.
+ */
+export type MyComponentClassKey = "root" | "header" | "typography" | "icon" | "children" | "hasShadow";
+
+type RootProps = Pick<MyComponentProps, "shadow">;
```

Comment:
Shouldn't we call this `MyComponentOwnerState`? (like MUI does [here](https://mui.com/material-ui/guides/creating-themed-components/#typescript))

Reply:
Good point. Now that I think about it, a single `ownerState` for the component instead of individual props for every styled-component/slot might make things simpler. 

I will discuss this with some other devs. 
This might make sense, not just for the comet-admin but for our projects in general.